demo/site/src/documents/pages/blocks/ProductListBlock.tsx

```diff
@@ -0,0 +1,20 @@
+import { type PropsWithData, withPreview } from "@comet/site-nextjs";
+import { type ProductListBlockData } from "@src/blocks.generated";
+import { PageLayout } from "@src/layout/PageLayout";
+
+export const ProductListBlock = withPreview(
+    ({ data: { products } }: PropsWithData<ProductListBlockData>) => {
+        return (
+            <PageLayout grid>
+                <ul>
+                    {(Array.isArray(products) ? products : [products]).map((product) => (
```

Comment:
This isn't correct. The generated type should be an array, but it isn't:

```
export interface ProductListBlockData {
    products: "cap" | "shirt" | "tie";
}
```

We need to support arrays here: https://github.com/vivid-planet/comet/blob/support-enum-arrays-in-block-field-decorator/packages/cli/src/commands/generate-block-types.ts#L34-L36.